### PR TITLE
fix(splash-screen): launchAutoHide not working on iOS

### PR DESCRIPTION
**Pull request by jcesarmobile**
launchAutoHide was not working on iOS

also the PR changes the default splash value from 0 to 500ms so the splash always shows

and update the docs accordingly

closes https://github.com/ionic-team/capacitor-plugins/issues/270
closes https://github.com/ionic-team/capacitor-plugins/issues/317
closes https://github.com/ionic-team/capacitor-plugins/issues/300

### CODE REVIEW

**Comment by IljaDaderko** [comment]
Same behaviour happens on android with fullscreen and immersive flags set to true, screen hides instantly without waiting for `.hide()` execution. Removing these 2 flags makes it work as expected.

```
launchAutoHide: false,
splashFullScreen: true,
splashImmersive: true
```

File: splash-screen/README.md [hunk 1]
```diff
@@ -31,13 +31,12 @@ SplashScreen.show({
 
 ## Hiding the Splash Screen
 
-By default, the Splash Screen is set to automatically hide after a certain amount of time (3 seconds). However, your app should boot much faster than this!
+By default, the Splash Screen is set to automatically hide after a certain amount of time (500 ms).
```
Comment (ikeith) [inline splash-screen/README.md:34]:
```suggestion
By default, the Splash Screen is set to automatically hide after 500 ms.
```

File: splash-screen/README.md [hunk 2]
```diff
@@ -31,13 +31,12 @@ SplashScreen.show({
...
 
-To make sure you provide the fastest app loading experience to your users, you should hide the splash screen when your app is ready to be used by calling `hide()` as soon as possible.
+If you want to be sure the splash never hides before the app is fully loaded, set `launchAutoHide` to `false`. You should hide the splash screen when your app is ready to be used by calling `hide()` as soon as possible.
```
Comment (ikeith) [inline splash-screen/README.md:36]:
```suggestion
If you want to be sure the splash screen never disappears before your app is ready, set `launchAutoHide` to `false`; the splash screen will then stay visible until manually hidden. For the best user experience, your app should call `hide()` as soon as possible.
```

File: splash-screen/README.md [hunk 3]
```diff
@@ -31,13 +31,12 @@ SplashScreen.show({
...
 
-If your app needs longer than 3 seconds to load, configure the default duration by setting `launchShowDuration` in your [Capacitor configuration file](https://capacitorjs.com/docs/config).
+If your you want to show the splash for a fixed amount of time, configure the default duration by setting `launchShowDuration` in your [Capacitor configuration file](https://capacitorjs.com/docs/config).
```
Comment (ikeith) [inline splash-screen/README.md:38]:
```suggestion
If, instead, you want to show the splash screen for a fixed amount of time, set `launchShowDuration` in your [Capacitor configuration file](https://capacitorjs.com/docs/config).
```

**Review by ikeith — APPROVED** [review]
Some minor suggestions to the documentation.

**Pull request merged**